overlays/tools/get-changed-modules-pull-request.sh

```diff
@@ -0,0 +1,185 @@
+note() { printf "\n${underline}${bold}${blue}Note:${reset} ${blue}%s${reset}\n" "$@" ; }
+
+#-----------------------------------------------------------------------------------------                   
+# Functions
+#-----------------------------------------------------------------------------------------                   
+function usage {
+    info "Syntax: get-changed-modules-pull-request.sh [OPTIONS]"
+    cat << EOF
+Options are:
+-h | --help : Display this help text
+--pr-number The number of the Pull Request
+EOF
+}
+
+module_names=(\
+    "buildutils" \
+    "wrapping" \
+    "gradle" \
+    "maven" \
+    "framework" \
+    "extensions" \
+    "managers" \
+    "obr" \
+)
+
+#-----------------------------------------------------------------------------------------                   
+# Process parameters
+#-----------------------------------------------------------------------------------------
+pr_number=""
+while [ "$1" != "" ]; do
+    case $1 in
+        -h | --help )       usage
+                            exit
+                            ;;
+
+        --pr-number )       pr_number="$2"
+                            shift
+                            ;;
+
+        * )                 error "Unexpected argument $1"
+                            usage
+                            exit 1
+    esac
+    shift
+done
+
+#-----------------------------------------------------------------------------------------                   
+# Functions
+#-----------------------------------------------------------------------------------------  
+
+function get_paths_changed_in_pr() {
+
+    h1 "Getting the file paths changed in Pull Request number ${pr_number}" 
+
+    # Extract changed module names from changed files from GitHub CLI output
+    mapfile -t changed_files_in_pr < <(gh pr diff --repo galasa-dev/galasa ${pr_number} --name-only)
+
+    h2 "Files changed:"
+
+    modules_changed_in_pr=()
+    for changed_file in "${changed_files_in_pr[@]}"; do
+        echo "$changed_file"
+        module=$(echo "$changed_file" | cut -d'/' -f2)
+        modules_changed_in_pr+=("$module")
+    done
+
+    # Remove possible duplicates from array of changed modules
+    declare -A unique_module_map
+
+    unique_modules_found_in_pr=()
+    for module in "${modules_changed_in_pr[@]}"; do
+    if [[ -z "${unique_module_map[$module]}" ]]; then
+        unique_modules_found_in_pr+=("$module")
+        unique_module_map[$module]=1
+    fi
+    done
+
+    h2 "Modules changed:"
+    echo "${unique_modules_found_in_pr[@]}"
+}
+
+function get_changed_modules_and_set_in_environment() {
+
+    h1 "Finding changed modules and setting environment variables that can be used in the GitHub Actions workflows..."
+
+    for module in "${unique_modules_found_in_pr[@]}"; do
+        if [[ "$module" == "buildutils" ]]; then
+            echo "BUILDUTILS_CHANGED=true" >> $GITHUB_OUTPUT
+            continue
+        fi
+        if [[ "$module" == "wrapping" ]]; then
```

Comment:
The env var could be built dynamically using the module name ? So adding or removing a module need not require changes to this script in future.